Using 'Peri' fails during compilation

Source:

```rust
#[resource_group(no_aliases)]
struct CrcEngineResources {
    crc_engine: CRC,
}
```

Compiler Output:

```
error[E0106]: missing lifetime specifier
  --> src/lib.rs:28:1
   |
28 | #[resource_group(no_aliases)]
   | ^ expected named lifetime parameter
   |
   = note: this error originates in the attribute macro `resource_group` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider introducing a named lifetime parameter
   |
28 ~ #'a, [resource_group(no_aliases)]
29 ~ struct CrcEngineResources<'a> {
```

The suggested 'fix' from the compiler doesn't appear to be valid syntax, or at least it's not valid in Rust 1.86.0 which is what I'm using.

If the macro syntax allows for it, changing `field.ty = parse_quote! { #peri_path<#ty> };` to `field.ty = parse_quote! { #peri_path<'static, #ty> };` will probably fix it. I suspect the dummy `Peri` in the tests will have to include a lifetime parameter in order to expose this while compiling the tests.

--- a/embedded-resources/src/lib.rs
+++ b/embedded-resources/src/lib.rs
@@ -188,7 +188,7 @@ pub fn resource_group(args: TokenStream, item: TokenStream) -> TokenStream {
 
     s.fields.iter_mut().for_each(|field| {
         let ty = &field.ty;
-        field.ty = parse_quote! { #peri_path<#ty> };
+        field.ty = parse_quote! { #peri_path<'static, #ty> };
     });
 
     quote! {

--- a/tests/src/dummy.rs
+++ b/tests/src/dummy.rs
@@ -1,9 +1,12 @@
 #![allow(non_camel_case_types)]
 #![allow(non_snake_case)]
 
-pub struct Peri<P> {
+use core::marker::PhantomData;
+
+pub struct Peri<'a, P> {
     #[expect(unused)]
     p: P,
+    _lifetime: PhantomData<&'a mut P>,
 }
 
 pub(crate) struct PA2;
@@ -15,25 +18,46 @@ pub(crate) struct TIM2;
 pub(crate) struct USB_OTG_FS;
 
 pub(crate) struct Peripherals {
-    pub(crate) PA2: Peri<PA2>,
-    pub(crate) PA3: Peri<PA3>,
-    pub(crate) PA4: Peri<PA4>,
-    pub(crate) PA11: Peri<PA11>,
-    pub(crate) PA12: Peri<PA12>,
-    pub(crate) TIM2: Peri<TIM2>,
-    pub(crate) USB_OTG_FS: Peri<USB_OTG_FS>,
+    pub(crate) PA2: Peri<'static, PA2>,
+    pub(crate) PA3: Peri<'static, PA3>,
+    pub(crate) PA4: Peri<'static, PA4>,
+    pub(crate) PA11: Peri<'static, PA11>,
+    pub(crate) PA12: Peri<'static, PA12>,
+    pub(crate) TIM2: Peri<'static, TIM2>,
+    pub(crate) USB_OTG_FS: Peri<'static, USB_OTG_FS>,
 }
 
 impl Peripherals {
     pub(crate) const fn new() -> Self {
         Self {
-            PA2: Peri { p: PA2 },
-            PA3: Peri { p: PA3 },
-            PA4: Peri { p: PA4 },
-            PA11: Peri { p: PA11 },
-            PA12: Peri { p: PA12 },
-            TIM2: Peri { p: TIM2 },
-            USB_OTG_FS: Peri { p: USB_OTG_FS },
+            PA2: Peri::<'static> {
+                p: PA2,
+                _lifetime: PhantomData,
+            },
+            PA3: Peri::<'static> {
+                p: PA3,
+                _lifetime: PhantomData,
+            },
+            PA4: Peri::<'static> {
+                p: PA4,
+                _lifetime: PhantomData,
+            },
+            PA11: Peri::<'static> {
+                p: PA11,
+                _lifetime: PhantomData,
+            },
+            PA12: Peri::<'static> {
+                p: PA12,
+                _lifetime: PhantomData,
+            },
+            TIM2: Peri::<'static> {
+                p: TIM2,
+                _lifetime: PhantomData,
+            },
+            USB_OTG_FS: Peri::<'static> {
+                p: USB_OTG_FS,
+                _lifetime: PhantomData,
+            },
         }
     }
 }

